Mathlib/LinearAlgebra/Isomorphisms.lean

```diff
@@ -60,67 +62,92 @@ theorem quotKerEquivRange_apply_mk (x : M) :
 #align linear_map.quot_ker_equiv_range_apply_mk LinearMap.quotKerEquivRange_apply_mk
 
 @[simp]
-theorem quotKerEquivRange_symm_apply_image (x : M) (h : f x ∈ f.range) :
+theorem quotKerEquivRange_symm_apply_image (x : M) (h : f x ∈ LinearMap.range f) :
     f.quotKerEquivRange.symm ⟨f x, h⟩ = f.ker.mkQ x :=
   f.quotKerEquivRange.symm_apply_apply (f.ker.mkQ x)
 #align linear_map.quot_ker_equiv_range_symm_apply_image LinearMap.quotKerEquivRange_symm_apply_image
 
+/-
+Porting note: In these definitions and theorems, a match to `Submodule.instModuleSubtype
+  MemSubmoduleInstMembershipInstSetLikeSubmoduleInstAddCommMonoidSubtypeMemSubmoduleInstMembership
+  InstSetLikeSubmodule` which takes 0.1s happens a few hundreds times, so `maxHeartbeats` and
```

Comment:
I wonder if giving this a less outrageously long name would speed up hash lookups at all (marginally)...

Reply:
#2967